Doc 1671/archive v1 onboarding docs to preview as reference for v2 gap analysis (#607)
swan-io/docs: docs/preview/new-onboarding/legacy-onboarding/index.mdx

---
title: Legacy onboarding reference
---

# Legacy onboarding reference

:::caution Deprecated

> These pages document the deprecated onboarding API. They are preserved as a read-only reference for gap analysis during the migration. Refer to [account onboarding enhancements](../index.mdx) for the current API.

Reviewer: Partners won't know what a gap analysis is and why they'd want to do that. Needs to be more plain english: 

This page documents the previous onboarding API (V1). For the current API, refer to [Account onboarding enhancements]."

Same comment as earlier, but V1 is the active API for partners who haven't migrated yet. Until June 2026, this is their implementation guidance. A partner reading this will think: "So where do I go?" and contact support.

Author: Thanks for the feedback. The "gap analysis" wording is internal; agreed, removing it. On the broader concern: this section is intentionally scoped as a reference archive, not primary implementation guidance. Hadrien's original request (Slack, 23 March) explicitly asked for v1 docs to be preserved in preview for partners still on v1, alongside the v2 diff. The deprecation callout links directly to the current API. Partners who haven't migrated yet should be directed here by their PIM, not landing here cold. That said, I'm happy to strengthen the callout text if you have a specific wording in mind.

Reviewer: Yes, the call-outs need to be stronger because this solution doesn't solve for: 
1. Partners on V1 who need help with their current integration. 
2. Partners moving from V1 to V2 - to understand the differences and what they have to do. With this solution, they'd have to move between two pages. 

This approach solves for a PIM taking the Partner through, but for Partners without a PIM, they'll be stuck and raise a support request. 

What is missing here is what the Partners need to do to migrate. We show what's changed but not necessarily the how - similar to this: https://docs.adyen.com/platforms/onboard-users/onboarding-versions?tab=api-only_onboarding_2_3#update-version. We need a migration guide that covers - does this live somewhere?
 
1. What's changing — mutation names, input structure (flat → nested), new required fields, response types
2. What happens to existing onboardings — can they coexist? Do v1 onboardings in progress need to be completed on v1?
3. Step-by-step migration path — update mutations → update input shapes → test in Sandbox → switch in Live
4. Deprecation timeline — when will v1 mutations be removed?
5. Testing checklist — key scenarios to validate

Author: This is right and important. It's out of scope for DOC-1671, which is purely the archive migration, but I've opened [DOC-1693](https://linear.app/swan/issue/DOC-1693/create-a-how-to-guide-for-onboarding-migration) to track it. The guide will build on top of DOC-1669 (which covers the technical diff) and add the partner-facing layer: step-by-step migration path, coexistence rules, and deprecation timeline. I'm aligning with Maxime on the two missing pieces before scoping the full work. The callout on this page will link to it once it's live.

> These pages document the deprecated onboarding API. They are preserved as a read-only reference for gap analysis during the migration. Refer to [account onboarding enhancements](../index.mdx) for the current API.

Author: ```suggestion
These pages document the deprecated onboarding API. They are preserved as a read-only reference to help you with the migration. Refer to [account onboarding enhancements](../index.mdx) for the current API.
```
